### ISSUE TITLE
Processing does not start if Android Mode is out of date

### ISSUE THREAD

**Issue by Haugaarden** [processing/processing#4467]
When I try to run any of the Processing 3.X versions, I'm prompted with an alert that looks like this:

![image](https://cloud.githubusercontent.com/assets/19310203/15187966/aa839508-17a3-11e6-8325-4573df2b9e66.png)

I can never get past this screen and into the IDE, as the splash screen just disappears and the IDE never shows up. The only mode I have in the 'modes' folder is the default java mode.

**Suggestion:** Make Processing open up in the default java mode, when there is a problem with the mode it is trying to open.

**Comment by benfry** [comment]
Your suggestion has already been implemented, so that's a bug that it's not working properly. 

However, where do you have Android Mode installed? Can you provide a copy of the Android Mode folder that's causing this to break?

**Comment by Haugaarden** [comment]
I only have Android Mode in my Processing 2.2.1 'modes' folder. I can open Processing 2, but every time I try to open any of the Processing 3 versions it will give me this alert. I might have tried to bring Android Mode to one of the early versions of Processing 3, and since then I have not been able to open Processing 3. 

There is only the default java mode in my Processing 3.1 'modes' folder, and I have not messed with the 'modes' folder since I downloaded Processing 3.1 from the Processing page. 

I'm guessing that it is giving me this alert because I tried to insert the Processing 2 Android Mode into one of the early versions of Processing 3, and is still thinking that I am trying to open an out of date Android Mode.

**Comment by benfry** [comment]
That sounds like you're using the old sketchbook folder with Processing 3, so it's picking up that mode. 

But in any case, try launching Processing this way:

Open Applications → Utilities → Terminal.app

Assuming you have Processing installed in Applications, type: 
```
cd /Applications
./Processing.app/Contents/MacOS/Processing 
```
What messages does it print?

**Comment by Haugaarden** [comment]
@benfry 
I am using Windows so I would have to use the CMD. Do you know what I should write then?

Also I think that I told it not to share Processing 2 and Processing 3 folders back when it asked me that, but I am not sure.

**Comment by benfry** [comment]
Open up the command prompt, cd to the place you've installed Processing, and enter:
```
.\java\bin\java -cp lib/pde.jar;core/library/core.jar;lib/jna.jar;lib/jna-platform.jar;lib/antlr.jar;lib/ant.jar;lib/ant-launcher.jar processing.app.Base
```

**Comment by Haugaarden** [comment]
@benfry 
It gave me this alert:

![image](https://cloud.githubusercontent.com/assets/19310203/15195052/5c0148bc-17c6-11e6-8958-5bf71e3100b9.png)

And it opened this webpage: https://github.com/processing/processing/wiki/Supported-Platforms

**Comment by Haugaarden** [comment]
This was the place I had set as cd:

![image](https://cloud.githubusercontent.com/assets/19310203/15195380/ec7e821e-17c7-11e6-8032-7e53a63bb70a.png)

**Comment by Haugaarden** [comment]
And this is what I got when I used the path of my Processing 3 folder that I extracted to my desktop. There is a message in the CMD but that doesn't tell much I think.

![image](https://cloud.githubusercontent.com/assets/19310203/15195558/af94693a-17c8-11e6-9d9a-7e9b34804fe7.png)

**Comment by benfry** [comment]
Is the date on your machine set to the year 2045?

**Comment by Haugaarden** [comment]
Not at all. It's set to the correct time, so I don't know why it says that.

![image](https://cloud.githubusercontent.com/assets/19310203/15195902/3c083814-17ca-11e6-8ea9-f3d58b51ebdf.png)

**Comment by benfry** [comment]
Ok, so that's an error coming from inside the Android Mode you have installed. Move it out of the modes folder and see what happens. (The 2045 date is the expiration date of the Android Debug certificate.)

But keep a copy of that folder—I'm still trying to figure out why it's able to break the rest of the app.

**Comment by Haugaarden** [comment]
I removed the Android Mode from: D:\Users\Martin\Documents\Processing\modes

And then I tried to start Processing 3. I was greeted with this alert:
![image](https://cloud.githubusercontent.com/assets/19310203/15196394/4acacfa4-17cc-11e6-9c70-310177e81cd1.png)

And after that I got the "Mode out of date" alert.

**Comment by Haugaarden** [comment]
I just had a breakthrough. I found another place where there was an Android Mode.
Found it here: D:\Users\Martin\Documents\modes
I deleted Android Mode and Processing 3 now opens as it should!

I will try to put my other Android Mode back where it used to be and see if it will still work.

**Comment by Haugaarden** [comment]
Processing 3 is working with the Processing 2 AndroidMode in the 'modes' folder where it was before.

![image](https://cloud.githubusercontent.com/assets/19310203/15196694/f2584d22-17cd-11e6-8805-634a19e16568.png)

My problem might have been because I had two Android Modes on my pc, one connected to Processing 2 and one connected to Processing 3.

**Comment by Haugaarden** [comment]
I downloaded the AndroidMode through the Contribution Manager, and it placed the new AndroidMode where the old one that I just deleted had been. So I had an AndroidMode for Processing 2, a place where Processing 3 was looking for modes. This have probably not been a bug but a mistake made by me. I apologize for the time I might have wasted.

You are free to close this issue, unless you suspect that there is a bug. I'm ready to answer questions if you have any. Thank you for your time @benfry

**Comment by benfry** [comment]
Can you please post a zip of the AndroidMode folder that was causing this problem? The bug we need to fix is that a bad Mode shouldn't be able to prevent Processing 3 from running. A lot of effort has gone into making sure that's the case (so that I'm not chasing after issues like this one), but I've apparently missed one of the cases.

**Comment by Haugaarden** [comment]
I have made a zip of the AndroidMode I deleted to get Processing 3 working, but it won't allow me to upload the zip here, even though it says that zip is supported.

![image](https://cloud.githubusercontent.com/assets/19310203/15208962/bb1ab30c-182e-11e6-86fe-e264e98a4209.png)

**Comment by benfry** [comment]
Is there somewhere else you can post it (temporarily) for me?

**Comment by Haugaarden** [comment]
Here it is:

https://www.dropbox.com/sh/75zj5pm10d83wlu/AAAH3hPqciQMS39KCZU6JytEa?dl=0

**Comment by benfry** [comment]
Thank you! I'm able to reproduce the problem and am looking into a fix.

**Comment by Haugaarden** [comment]
Happy to hear that! Good luck.

**Comment by benfry** [comment]
Fixed for 3.1.1.
